pkg/binding/bindingpolicy.go

```diff
@@ -101,6 +151,25 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 	return c.deleteResolutionForBindingPolicy(ctx, bindingPolicyName)
 }
 
+// identifyMissingStatusCollectors finds all missing StatusCollector objects for bp.
+// If an error other than "not found" occurs, identifyMissingStatusCollectors returns a nil slice and the error;
+// otherwise, identifyMissingStatusCollectors returns a slice of StatusCollector object names and a nil error.
+func (c *Controller) identifyMissingStatusCollectors(bp *v1alpha1.BindingPolicy) ([]string, error) {
+	statusCollectorNames := sets.NewString([]string{}...)
```

Comment:
https://github.com/kubernetes/apimachinery/blob/v0.29.12/pkg/util/sets/string.go#L21

It would also be less verbose to simply `sets.New[string]()`.

Reply:
Improved.